### ISSUE TITLE
[beta] SL fails to load settings-schema.json (Windows?)

### ISSUE THREAD

**Issue by Gliptal** [SublimeLinter/SublimeLinter#776]
Woke up this morning with SL4 not working anymore: no matter what the linter or the file, I get a number of AttributeError(s), mostly `AttributeError: 'NoneType' object has no attribute 'pop'`, `AttributeError: 'Daemon' object has no attribute '_callback'`, `AttributeError: 'NoneType' object has no attribute 'get_view_dict'`, and `AttributeError: 'NoneType' object has no attribute 'get_line_dict'`. These errors repeat themselves whenever I modify or save the file that's supposedly being linted.

Windows 10 1709 (16299.192)
Sublime Text 3 (3143)
SublimeLinter v4.0.0-rc.9.1

````
DPI scale: 1.25
startup, version: 3143 windows x64 channel: stable
executable: /C/Program Files (x86)/Portable/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files (x86)/Portable/Sublime Text 3
packages path: /C/Program Files (x86)/Portable/Sublime Text 3/Data/Packages
state path: /C/Program Files (x86)/Portable/Sublime Text 3/Data/Local
zip path: /C/Program Files (x86)/Portable/Sublime Text 3/Packages
zip path: /C/Program Files (x86)/Portable/Sublime Text 3/Data/Installed Packages
ignored_packages: ["ActionScript", "AppleScript", "ASP", "C#", "Clojure", "Color Scheme - Legacy", "D", "Diff", "Erlang", "Go", "Graphviz", "Groovy", "Haskell", "Lisp", "Matlab", "Objective-C", "OCaml", "Pascal", "Perl", "Rails", "RestructuredText", "Ruby", "Rust", "Scala", "ShellScript", "TCL", "Textile", "Vintage"]
Errors parsing theme:
icon_folder is missing layer0.opacity, setting to 1.0 for backwards compatibility
icon_folder_loading is missing layer0.opacity, setting to 1.0 for backwards compatibility
pre session restore time: 0.232949
startup time: 0.300949
first paint time: 0.300949
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.02-bz2
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin AdvancedNewFile.AdvancedNewFile
reloading plugin Alignment.Alignment
reloading plugin All Autocomplete.all_views_completions
reloading plugin BufferScroll.BufferScroll
reloading plugin GitGutter.plugin
reloading plugin neko nml syntax.neko-nml-std-completions
reloading plugin Open Folder.OpenFolder
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin Stylus.Stylus
reloading plugin SublimeLinter-contrib-pug-lint.linter
reloading plugin SublimeLinter-contrib-stylint.linter
reloading plugin SublimeLinter-cpplint.linter
reloading plugin SublimeLinter-csslint.linter
reloading plugin SublimeLinter-jshint.linter
reloading plugin SublimeLinter-pycodestyle.linter
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin Babel.Babel
reloading plugin CSScomb.CSScomb
reloading plugin SublimeLinter-contrib-eslint.linter
plugins loaded
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 210, in on_api_ready
    m.plugin_loaded()
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 46, in plugin_loaded
  File "lint.settings in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 63, in load
  File "lint.settings in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 139, in on_update
  File "lint.settings in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 231, in settings_valid
  File "lint.util in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 667, in load_json
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime.py", line 192, in load_resource
    raise IOError("resource not found")
OSError: resource not found
Package Control: Skipping automatic upgrade, last run at 2018-01-08 10:37:59, next run at 2018-01-08 11:37:59 or after
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 42, in on_activated_async
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 85, in get_we_count
AttributeError: 'NoneType' object has no attribute 'get_view_dict'
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 102, in on_activated_async
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 317, in check_syntax
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 323, in clear
  File "lint.linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 1194, in clear_view
AttributeError: 'NoneType' object has no attribute 'pop'
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 42, in on_activated_async
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 85, in get_we_count
AttributeError: 'NoneType' object has no attribute 'get_view_dict'
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 389, in run_callback
    expr()
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 620, in <lambda>
    run_callback('on_hover', callback, lambda: callback.on_hover(v, point, hover_zone))
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 171, in on_hover
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 420, in open_tooltip
AttributeError: 'NoneType' object has no attribute 'get_line_dict'
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 42, in on_activated_async
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 85, in get_we_count
AttributeError: 'NoneType' object has no attribute 'get_view_dict'
SublimeLinter: computed PATH from system:
C:\Users\Utente\AppData\Roaming\npm
C:\Program Files (x86)\Python\Scripts\
C:\Program Files (x86)\Python\
C:\Program Files (x86)\Intel\iCLS Client\
C:\Program Files\Intel\iCLS Client\
C:\Windows\system32
C:\Windows
C:\Windows\System32\Wbem
C:\Windows\System32\WindowsPowerShell\v1.0\
C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL
C:\Program Files\Intel\Intel(R) Management Engine Components\DAL
C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT
C:\Program Files\Intel\Intel(R) Management Engine Components\IPT
C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common
C:\Program Files\Intel\WiFi\bin\
C:\Program Files\Common Files\Intel\WirelessCommon\
C:\WINDOWS\system32
C:\WINDOWS
C:\WINDOWS\System32\Wbem
C:\WINDOWS\System32\WindowsPowerShell\v1.0\
C:\Program Files (x86)\Git\cmd
C:\Program Files (x86)\NML
C:\Users\Utente\AppData\Local\Microsoft\WindowsApps

 
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 102, in on_activated_async
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 317, in check_syntax
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 323, in clear
  File "lint.linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 1194, in clear_view
AttributeError: 'NoneType' object has no attribute 'pop'
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 42, in on_activated_async
  File "status_bar_view in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 85, in get_we_count
AttributeError: 'NoneType' object has no attribute 'get_view_dict'
Traceback (most recent call last):
  File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 111, in on_activated_async
  File "sublime_linter in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 299, in hit
  File "lint.queue in C:\Program Files (x86)\Portable\Sublime Text 3\Data\Installed Packages\SublimeLinter.sublime-package", line 19, in hit
AttributeError: 'Daemon' object has no attribute '_callback'
````

**Comment by braver** [comment]
You probably (hopefully) also got a popup from package control explaining that a dependency was installed and you should restart because plugins may not work correctly.

**Comment by Gliptal** [comment]
Nothing from Package Control, I already restarted both ST3 and the OS to no avail.

**Comment by braver** [comment]
Can you run the "satisfy dependencies" command from package control and then restart?

**Comment by Gliptal** [comment]
Ran, getting similar tracebacks.

**Comment by braver** [comment]
It seems something about your portable setup prevents it from loading a json file that we ship. I cannot reproduce this in a normal setup (albeit on a Mac).

**Comment by Gliptal** [comment]
Would it be this error? `File "C:\Program Files (x86)\Portable\Sublime Text 3\sublime.py", line 192, in load_resource
    raise IOError("resource not found")`

**Comment by braver** [comment]
Yes, that would be my primary concern.

**Comment by Gliptal** [comment]
I'll see what I can come up with. The file is there.

**Comment by Gliptal** [comment]
@braver Solved, in case editing the previous post didn't notify you.

**Comment by braver** [comment]
Thanks, I wasn't notified about the edit. Glad you were able to fix it. I'm not really familiar with running ST portably, is there something we can do here to make it work out of the box?

**Comment by Gliptal** [comment]
I honestly don't know. The curious thing is that everything was working the day before, I have no clue why it stopped working.

It's the only SublimeLinter file that seems to need to be unpacked and placed into `Data\Packages,` is this as intended? Maybe use the packed version instead?

**Comment by braver** [comment]
We pushed an update this weekend that uses that json file to validate your settings. It's a new feature and a new file. I guess the api to read that file doesn't work in your portable setup.

**Comment by Gliptal** [comment]
Fair enough, until someone else has the same issue I wouldn't spend time trying to replicate this.

**Comment by braver** [comment]
Ok, when it happens again, we'll look into it.

**Comment by AaronAsAChimp** [comment]
I'm also having this problem. I am, however, not using a portable version of Sublime Text.

I *have* tried restarting Sublime and running "Satisfy dependencies" neither resolved the issue. @Gliptal's fix worked for me.

Windows 10 1709 (16299.192)
Sublime Text 3 (3156)
SublimeLinter: v4.0.0-rc.9.2

```
DPI scale: 1
startup, version: 3156 windows x64 channel: dev
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/aaronasachimp/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/aaronasachimp/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/aaronasachimp/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.166024
startup time: 0.308024
first paint time: 0.313024
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors
reloading plugin Default.comment
reloading plugin Default.convert_color_scheme
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin ShellScript.ShellScript
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin Bootstrap 4 Autocomplete.bootstrap_4_autocomplete
reloading plugin DocBlockr.jsdocs
reloading plugin EditorConfig.EditorConfig
reloading plugin FileDiffs.file_diffs
reloading plugin Flow.Flow
reloading plugin Jekyll.jekyll
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin PackageDev._logging
reloading plugin PackageDev.main
reloading plugin SublimeLinter-javac.linter
reloading plugin SublimeLinter-jshint.linter
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin TortoiseSVN.TortoiseSVN
reloading plugin Wrap Plus.wrap_plus
reloading plugin Babel.Babel
reloading plugin HTML-CSS-JS Prettify.HTMLPrettify
reloading plugin Pretty JSON.PrettyJson
reloading plugin Pretty JSON.PrettyJsonListeners
plugins loaded
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 210, in on_api_ready
    m.plugin_loaded()
  File "sublime_linter in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 46, in plugin_loaded
  File "lint.settings in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 63, in load
  File "lint.settings in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 139, in on_update
  File "lint.settings in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 231, in settings_valid
  File "lint.util in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 667, in load_json
  File "C:\Program Files\Sublime Text 3\sublime.py", line 192, in load_resource
    raise IOError("resource not found")
OSError: resource not found
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 487, in on_post_save_async
    callback.on_post_save_async(v)
  File "sublime_linter in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 131, in on_post_save_async
  File "sublime_linter in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 453, in file_was_saved
  File "sublime_linter in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 317, in check_syntax
  File "sublime_linter in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 323, in clear
  File "lint.linter in C:\Users\aaronasachimp\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 1194, in clear_view
AttributeError: 'NoneType' object has no attribute 'pop'
[...]
```

**Comment by braver** [comment]
Could it perhaps be that the / makes it fail to load under Windows?

**Comment by AaronAsAChimp** [comment]
... yes and no. It is the slashes, but its not the forward slashes. Its the `os.path.join()`. Joining the path with all forward slashes fixes it.

**Comment by Xunnamius** [comment]
I was also experiencing this problem like the others. Windows 10, normal non-portable ST3. Even completely removed SL and all associated files/folders.

When I first started ST3, I got a whole slew of strange errors including one about downloading some sort of json reading package after ST3 attempted `reloading plugin SublimeLinter.busy_indicator_view`. I'm guessing that was a missing dependency that got installed. Restarting ST3 turned those errors into other new and distinct errors.

@Gliptal's solution almost worked for me. Unfortunately, `C:\Users\<USER>\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\Resources` did not exist because `C:\Users\<USER>\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter` was never created. After creating that path, I downloaded the `settings-schema.json` file from Github and placed it there.

Now things are working! Thanks again for all the hard work, guys.
